### ISSUE TITLE
Hangs on splash screen when running release version

### ISSUE THREAD

**Issue by toadlyBroodle** [toadlyBroodle/Sublist#1]
When running release version (as opposed to debug version), app hangs on splash screen. Debug build option "Compile with .NET Native tool chain" appears to be the culprit. Debug build option "Optimize code" seems to compile fine without compiling with .NET Native tool chain.

http://stackoverflow.com/questions/33004675/uwp-app-release-version-hangs-on-splash-screen

**Comment by seboschtion** [comment]
So, it looks like the problem is caused by the `GetFileAsync`\- Method (http://stackoverflow.com/questions/33004675/uwp-app-release-version-hangs-on-splash-screen).
I tested the solution, but then the items did not get saved (the state of the eye-icon, however, did). But I'm sure you already know that.

My question is: Why don't save all the data in a SQLite database? I don't know, if you can expect some improvements, but when you allow me to, I will try it.
At least: Data handling would be cleaner and faster.

**Comment by clankwright** [comment]
Yeah I noticed the same thing. I'm just learning .NET and c# so I'm not aware of how to save in a SQLite database, would that fix the folder accessibility issues I've been having? So far this is the first way to save data that I've found that works (not inside the Task.Run()). Feel free to modify the code and send a pull request.

**Comment by seboschtion** [comment]
I hope this would fix the issue. We'll see. Do I have permissions to push to a feature branch and create a pull request?

**Comment by clankwright** [comment]
Cool. Yeah for sure, I added you as a collaborator. I haven't worked with github much yet, so not really sure how this usually works, but I set fix_splash_screen_hang as the base branch, so do you just fork it and then send pull requests, or just push directly to it?

**Comment by seboschtion** [comment]
(I haven't worked on an open source project yet, so I don't really know how it works, too ;-))

I created the pull request, but it should be merged into the master branch, I think. The reason: It still does not fix the problem. There are more steps needed. The problem is that I couldn't easily replace your storage solution with mine, because all handlers and the mainpage are constructed too complex.
Here is where the MVVM pattern comes in: You then can easily exchange parts of the code, and it continues to work. It takes some time to understand all advantages, but once you've worked on a project using it, you will see what that it simplifies a lot.
So you really should consider to implement MVVM, its still possible because the project is in an early stage. Of course I can help you with that, but it will change almost everything of your work, which I admire.

So, in conclusion, you choose whether you want to merge it or not. I can also create a new pull request which targets the fix_splash_screen_hang branch. Let me know where I can help you next. Or if I should not.

**Comment by clankwright** [comment]
Right on, thanks alot for adding in the database feature. I originally wanted to use a SQLite database to store the data, but was too lazy to learn how on top of everything else haha. Very much appreciated! I am fully on board with rewriting the project to use MVVM, because you're right, the project is already too complex which makes fixing bugs an ever increasing problem. I am still not sure how exactly to do this, but please feel free to do anything you want to the existing code. I'm tired of fighting with bugs in the existing structure! Over the next few days, I'll be learning about MVVM and will take a stab at rewriting the project if you haven't gotten to it by then. Thanks again for your help, it's awesome to finally work with someone besides just myself on a project, cheers!

**Comment by seboschtion** [comment]
Great, I'm happy that you're happy! So I'll send you another pull request but it can take a few days. I think this project is ideal to learn a lot about MVVM and SQLite :-) Not only you, me too. Nothing to thank for, I'm curious where this project leads to!
